File: HW2 submission/flip_flop.cpp
// Elana Lapins - Homework 2 Problem 0: Flip Flop
// Programming for Engineers, due: 09/17/2021

#include<iostream>
#include<string>

void main() {
	//INPUTS:
	int start = 0; // seconds to start at 
	int sec = 0; // # of seconds
	int a = 0; // a (flip)
	int b = 0; // b (flop)

	int cases = 0; 
	std::cin >> cases; //gets first value as num of cases

	for (int i = 0; i < cases; i++) {
		std::cout << "Case " << i << ":\n";

		for (int S = 0; S < cases; S += 3) {
			std::cin >> start;
		}
		for (int SEC = 1; SEC < cases; SEC += 3) {
			std::cin >> sec;
		}
		for (int A = 2; A < cases; A += 3) {
			std::cin >> a;
		}
		for (int B = 3; B < cases + 1; B+= 3) {
			std::cin >> b;
		}

		for (int w = start; w < start + sec; w++) {
			if (w% a == 0 && w % b == 0) {
				std::cout << "flipflop\n";
			}
			else if (w % a == 0) {
				std::cout << "flip\n";
			}
			else if (w % b == 0) {
				std::cout << "flop\n";
			}
			else {
				std::cout << w <<"\n";
			}
		}
		
	}
}

File: Liebniz_Flip_Money_Py/Flip_Flop.py
# homework 6: programming for Engineers
# Elana Lapins, 11/3/2021

#inputs: start second, number of seconds to simulate, a, b

#FLIP: on seconds divisible by a
#FLOP: on seconds divisible by b
#FLIPFLOP: on seconds divisible by both a and b
#print number: if not divisible by either

cases = int(input())

for i in range(cases):
    print("Case: ", i, "\n")

    list = input()
    split = list.split(" ")
    for i in range(0,len(split)):
        split[i] = int(split[i]) #convert to int list

    list = split;
    start = list[0]
    duration = list[1]
    a = list[2]
    b = list[3]

    for j in range(start,start+duration):
        if int(j)%a ==0:
            if int(j)%b==0:
                print("flipflop\n")
            else:
                print("flip\n")
        elif int(j)%b ==0:
            print("flop\n")
        else:
            print(j,"\n")
    

